### PR TITLE
[Owners Review] Added and updated some enums for DCPower

### PR DESCRIPTION
**Pull request by ni-pkhare**
### What does this Pull Request accomplish?

The proto file is updated with some more enums and the changes made to enums.py is documented in changes.md file.

### Why should this Pull Request be merged?

Following enums were added/modified :
- In 'OutputStates' : Following were added
    - NIDCPOWER_VAL_OUTPUT_OVER_VOLTAGE
    - NIDCPOWER_VAL_OUTPUT_OVER_CURRENT
    - NIDCPOWER_VAL_OUTPUT_UNREGULATED
  
- In 'PowerAllocationMode' : Following were renamed
    - NIDCPOWER_VAL_POWER_ALLOCATION_MODE_DISABLED to NIDCPOWER_VAL_DISABLED
    - NIDCPOWER_VAL_POWER_ALLOCATION_MODE_AUTOMATIC to NIDCPOWER_VAL_AUTOMATIC
    - NIDCPOWER_VAL_POWER_ALLOCATION_MODE_MANUAL to NIDCPOWER_VAL_MANUAL

The below change was reverted back.
 In 'ComplianceLimitSymmetry' : Following were renamed
    - NIDCPOWER_VAL_SYMMETRIC to NIDCPOWER_VAL_COMPLIANCE_LIMIT_SYMMETRY_SYMMETRIC
    - NIDCPOWER_VAL_ASYMMETRIC to NIDCPOWER_VAL_COMPLIANCE_LIMIT_SYMMETRY_ASYMMETRIC

### What testing has been done?

Manually verified.

### CODE REVIEW

File: source/codegen/metadata/nidcpower/CHANGES.md [hunk 1]
```diff
@@ -41,6 +41,13 @@ Following enums were defined :
 - NIDCPOWER_VAL_CANCEL
```
Comment (doshirohan) [inline source/codegen/metadata/nidcpower/CHANGES.md]: This is not your change, but looks like these two lines call out specific enum values. Can you replace these two lines with name of the enum `CalibrationCloseAction` instead?

**Review by doshirohan — APPROVED** [review]

File: source/codegen/metadata/nidcpower/CHANGES.md (hunk 1 above)
Comment (ni-pkhare) [reply source/codegen/metadata/nidcpower/CHANGES.md]: Done.

File: generated/nidcpower/nidcpower.proto [hunk 2]
```diff
@@ -376,8 +376,8 @@ enum CalibrationCloseAction {
 enum ComplianceLimitSymmetry {
   option allow_alias = true;
   COMPLIANCE_LIMIT_SYMMETRY_UNSPECIFIED = 0;
-  COMPLIANCE_LIMIT_SYMMETRY_NIDCPOWER_VAL_SYMMETRIC = 0;
-  COMPLIANCE_LIMIT_SYMMETRY_NIDCPOWER_VAL_ASYMMETRIC = 1;
+  COMPLIANCE_LIMIT_SYMMETRY_NIDCPOWER_VAL_COMPLIANCE_LIMIT_SYMMETRY_SYMMETRIC = 0;
```
Comment (jonlugoatx) [inline generated/nidcpower/nidcpower.proto]: This is quite long and redundant. Could provide some information as to why this change was done?

Comment (reckenro) [reply generated/nidcpower/nidcpower.proto]: It looks like it was updated to match then name in the header file (see [here](https://github.com/doshirohan/grpc-device/blob/features/ni-dcpower/imports/include/nidcpower.h#L457)). It's unfortunate the constant also has `COMPLIANCE_LIMIT_SYMMETRY` in the name since we also prepend the name here with the enum name.

I think the same is done for the PowerAllocationMode enum a few lines down (476), the enum value names are taken straight from the header file. That being said this does look redundant and maybe for these in enums.py we could leave out `COMPLIANCE_LIMIT_SYMMETRY` and `POWER_ALLOCATION_MODE` from the enum values' names.

Comment (doshirohan) [reply generated/nidcpower/nidcpower.proto]: This came out of discussion with Alex Dubois. You can find the discussion in [task description here](https://dev.azure.com/ni/DevCentral/_workitems/edit/1445386), but essentially we noticed that attribute values in the header file are named like this.
image

But our metadata refers to them as VAL_SYMMETRIC and VAL_ASYMMETRIC. Alex said that we should be using descriptive names. I didn't realize that we also prefix the value with enum name making it quite long. I think we can revert this part of the change and CHANGES.md.

**Review by reckenro — APPROVED** [review]
I think this change is fine and aligns with other enums / matches the header file. But consider if it makes sense to shorten the enums.py value names where it makes sense to reduce the redundancy in the generated enum value names in the .proto file.

File: source/codegen/metadata/nidcpower/CHANGES.md [hunk 3]
```diff
@@ -38,9 +38,15 @@ The following attributes were added :
 
 Following enums were defined :
 - InternalReference
-- NIDCPOWER_VAL_CANCEL
-- NIDCPOWER_VAL_COMMIT
+- CalibrationCloseAction
```
Comment (doshirohan) [inline source/codegen/metadata/nidcpower/CHANGES.md:41]: note: Line 41 is just fixing the change documentation. Instead of listing individual values from enum, we are now listing name of the enum itself.

File: generated/nidcpower/nidcpower.proto (hunk 2 above)
Comment (doshirohan) [reply generated/nidcpower/nidcpower.proto]: @reckenro definitely makes sense. We will remove this change, and fixup and document the `POWER_ALLOCATION_MODE` enum values.

Palash, make sure you update PR description as well to reflect this change.

**Review by jonlugoatx — APPROVED** [review]
As long as we address the comment about reverting the compliance limit enums.

File: generated/nidcpower/nidcpower.proto (hunk 2 above)
Comment (ni-pkhare) [reply generated/nidcpower/nidcpower.proto]: Done and updated the PR description. Will close this PR once build completes.

File: source/codegen/metadata/nidcpower/CHANGES.md (hunk 3 above)
Comment (ni-pkhare) [reply source/codegen/metadata/nidcpower/CHANGES.md:41]: Correct. So can I resolve this comment?

Comment (doshirohan) [reply source/codegen/metadata/nidcpower/CHANGES.md:41]: Yes, I just added it as note for owners so that they know what this change is.
